test/test.js

```diff
@@ -18,557 +19,655 @@ var describeStatusMessage = !/statusMessage/.test(http.IncomingMessage.toString(
   ? describe.skip
   : describe
 
-describe('finalhandler(req, res)', function () {
-  describe('headers', function () {
-    it('should ignore err.headers without status code', function (done) {
-      request(createServer(createError('oops!', {
-        headers: { 'X-Custom-Header': 'foo' }
-      })))
-        .get('/')
-        .expect(shouldNotHaveHeader('X-Custom-Header'))
-        .expect(500, done)
-    })
-
-    it('should ignore err.headers with invalid res.status', function (done) {
-      request(createServer(createError('oops!', {
-        headers: { 'X-Custom-Header': 'foo' },
-        status: 601
-      })))
-        .get('/')
-        .expect(shouldNotHaveHeader('X-Custom-Header'))
-        .expect(500, done)
-    })
+const servers = [
+  ['http', createHTTPServer],
+  ['http2', createHTTP2Server]
+]
+
+for (var i = 0; i < servers.length; i++) {
```

Comment:
I am not sure it matters, but one thing we could do to reduce the change set here would be to do the loop outside of the old code to not make it indent more. Something like this:


```
var topDescribe = function(createServer) { return function () { /* ... */} }
for (var i = 0; i < servers.length; i++) {
  describe(name, topDescribe(createServer));
}
```

Reply:
To remove the indent you would need that inner function to be named, but just wanted to do some example code to show how to write this without the whole file being marked as changed.